### ISSUE TITLE
Backing Up Bookmarks Not Working As Expected...

### ISSUE THREAD

**Issue by Geyup** [QupZilla/qupzilla#1807]
I had a previous question about how to backup, but I belive this is now either a bug, or the intentions of the exporting the bookmarks is not saving as it should.

I truly believe that when you export your bookmarks and then import them back, they should go back to exactly the same place as they were created...

Please look at the screen shots below.

1. This is how my bookmarks look after I exported and imported them back.

![qup1](https://cloud.githubusercontent.com/assets/13072581/11764444/507eac88-a128-11e5-9450-93211c06c99f.jpg)

2. This is how my original bookmarks looks, yes they are in the same order but they are in a different placement and this is where I expected them to end back up, not under this ```HTML Import``` section.
Now I have to manually go back and move everyone.

![qup2](https://cloud.githubusercontent.com/assets/13072581/11764449/8f625328-a128-11e5-8451-7ec0ad8e88ab.jpg)

Why would anyone want to go to all the trouble of placing bookmarks where I have them, which is what is normally done under any browser, and then have them imported into another section? I truly don't think anyone wants this, and if they do, or don't, either way, can we ```PLEASE``` have an option as to which way you want your bookmarks imported back?

thank you

**Comment by Geyup** [comment]
I found that starting QupZilla for the first time, then copying the old ```bookmarks.json``` and ```browsedata.db``` give the original results.

Without the ```browsedata.db``` the favicons don't appear, but is it safe to copy and use the original ```browsedata.db```?

These are the only two files I see needed to give the original results, should any other original files be copied over as well?

**Comment by innermous** [comment]
I am sorry, not related to discussion, but can you please share a link to these dark theme and icons?

**Comment by JHooverman** [comment]
I agree, the process of backing up & restoring existing bookmarks should be improved. It seems not to be reasonable for an "average" user to know and to look for the files `browsedata.db` and `bookmarks.json` to copy them back for backup/ restore purposes.
To my mind we will have to differentiate between three use cases:

(1) Initial import, when migrating from other browsers to QupZilla
The import procedure works as expected. However, the imported bookmarks need some manual sorting if required
(2) Export to html-file, when migrating from QupZilla to other browsers
The export procedure works as expected.
(3) Backup & Restore within QupZilla
Maybe we should add a Backup & Restore option in addition to Import & Export which will allow to restore the bookmarks in same structure as they have been backed up.

**Comment by innermous** [comment]
Can you explain what's wrong with browsedata.db? Isn't it just importer trouble?

**Comment by cranes-bill** [comment]
@JHooverman, can you scroll with the mouse scroller over the bookmarks area (of course if you have a long enough bookmarks list)? 
I can scroll with the arrows only over it, but not with mouse (TP) scroll. Is that a bug or ...? 
(Qt5 only)

**Comment by JHooverman** [comment]
@cranes-bill: I am afraid, but my bookmark-list is not long enough, so, no chance to test it.

**Comment by Geyup** [comment]
Don't forget the ```Width``` of the menu I mentioned, I didn't see any reply about that...

**Comment by Geyup** [comment]
@innermous well if you're a Linux user here ya go;

http://tinyurl.com/oonfy4a
http://tinyurl.com/oxeck3s

**Comment by innermous** [comment]
@Geyup it's strange, since I already have it as main gtk theme. But as I said before, my window manager doesn't treat qupzilla styles and icons serious. But thanks anyway

**Comment by Geyup** [comment]
What WM you using?

**Comment by innermous** [comment]
@Geyup awesome wm, I am currently working on my problem. Not sure it is related to thread :)

**Comment by Geyup** [comment]
Try LXAppearance

**Comment by innermous** [comment]
@Geyup not that easy.

**Comment by Geyup** [comment]
@innermous what distro?

**Comment by innermous** [comment]
@Geyup archlinux.

**Comment by Geyup** [comment]
By the way, when we are done with this conversation, please go back and delete all your replies pertainaing to this and I will do the same, since it's off topic...

Archlinux sheesh man that's easy; ```pacman -Syu lxappearance```

**Comment by innermous** [comment]
@Geyup dude, I have that package, it's more complicated.

**Comment by Geyup** [comment]
I've run awesome before, it has it's own menu, but beyond yet, using lxappearance, it's going to theme your GTK apps, so what's complicated about that?

**Comment by innermous** [comment]
@Geyup  I've found solution, preloading DESKTOP_SESSION="gnome".
I've also seen lx-session package, maybe it's just for that, not sure.

**Comment by Geyup** [comment]
Yes you might have to set ENV variable, I do the same in Openbox with a file called ```environment```

But I do this for QT5...

# Set the environment variable to force a specific style for QT5
export QT_STYLE_OVERRIDE=gtk

# Set desktop to use for QT5
export DESKTOP_SESSION=gnome

**Comment by innermous** [comment]
@Geyup ye, everything works now, thanks.

**Comment by Geyup** [comment]
Ok great delete all these threads, since off topic...

cheers

**Comment by nowrep** [comment]
It's intentional, because it is not *backup*, it is *export*.

**Comment by Geyup** [comment]
What reason(s) is someone going to export there bookmarks, if it's not for a backup then?

All I know is that the bookmarks should look the same way an end-user creates them when they export and import them back and this is not happening...

**Comment by JHooverman** [comment]
@nowrep: The issue is here, that when the bookmarks of QupZilla are exported and then imported again the structure of the imported bookmarks is different. Thus, they will have to be sorted again, which takes time again.

I would like to refer again to three different use cases:

(1) Initial import, when migrating from other browsers to QupZilla
The import procedure works as expected. However, the imported bookmarks need some manual sorting if required
(2) Export to html-file, when migrating from QupZilla to other browsers
The export procedure works as expected. However, importing them again [because the export function will be used for backup purposes] will cause sorting them again.

So maybe you will accept the following feature request, unless there is no work around for (2), use for backup:

**(3) Backup & Restore within QupZilla**
Maybe we should add a Backup & Restore option in addition to Import & Export which will allow to restore the bookmarks in same structure as they have been backed up.

**Comment by nowrep** [comment]
> (3) Backup & Restore within QupZilla
> Maybe we should add a Backup & Restore option in addition to Import & Export which will allow to restore the bookmarks in same structure as they have been backed up.

Yep
